app/livechat/server/api/v1/sns.js

```diff
@@ -0,0 +1,24 @@
+import { check } from 'meteor/check';
+
+import { API } from '../../../../api/server';
+import { SNS } from '../../lib/SNS';
+
+API.v1.addRoute('livechat/sns', {
+	async post() {
+		try {
+			check(this.bodyParams, {
+				room_id: String,
+				category: String,
+				action: String,
+				properties: Object,
+				event_type: String,
+			});
+
+			await SNS.sendEvent({ ...this.bodyParams, timestamp: new Date() });
+
+			return API.v1.success();
+		} catch (e) {
+			return API.v1.failure(e);
+		}
+	},
+});
```

Comment:
Should we add some kind of auth check based on visitor token or roomId or something else? So that we don't get spammed?

Reply:
Most of the Livechat Apis are unprotected, are you talking about something like if room Id is not valid throw error?

Reply:
Yes, something kind of that. This use case is different. Calling this endpoint will directly push data to the SNS topic. So we should add some validation if possible. i.e. transcript endpoint validates if the token and room are valid. We can discuss more in the meeting with other folks.

Reply:
Lets move forward without this for now.  We have another story to investigate solutions.